x-pack/plugins/apm/server/saved_objects/apm_indices.ts

```diff
@@ -15,6 +15,24 @@ export const apmIndices: SavedObjectsType = {
   mappings: {
     properties: {
       /* eslint-disable @typescript-eslint/naming-convention */
+      'xpack.apm.sourcemapIndices': {
+        type: 'keyword',
+      },
+      'xpack.apm.errorIndices': {
+        type: 'keyword',
+      },
+      'xpack.apm.onboardingIndices': {
+        type: 'keyword',
+      },
+      'xpack.apm.spanIndices': {
+        type: 'keyword',
+      },
+      'xpack.apm.transactionIndices': {
+        type: 'keyword',
+      },
+      'xpack.apm.metricsIndices': {
+        type: 'keyword',
+      },
       'apm_oss.sourcemapIndices': {
         type: 'keyword',
       },
```

Comment:
Can we run migrations on saved objects so we don't have to keep the old names?

Reply:
There is an example on this exact migration use case here: https://github.com/elastic/kibana/blob/master/src/core/server/saved_objects/migrations/README.md#example

Reply:
I'll try that, In the current iteration i had it support both new and old configs, and then migrate to new on index save, but it makes sense to do a proper migration supported by saved_objects.

Reply:
those docs were out of date, but i was able to use a migration function to accomplish this